### ISSUE TITLE
rhash 1.4.2 configure fails with macOS clang

### ISSUE THREAD

**Issue by alebcay** [rhash/RHash#169]
This issue was found in the process of packaging `rhash` 1.4.2 in Homebrew - see https://github.com/Homebrew/homebrew-core/pull/81385.

When building `rhash` 1.4.2 in macOS, the following error is encountered on the `configure` step:
```
Checking for target OS ... Darwin 

Error: compiler not found

Check "config.log" if you do not understand why it failed.
```

It appears that the logic for determining the compiler does not work on `clang` versions on macOS (at least for recent versions of the `clang` supplied by Apple's Xcode Command Line Tools). Specifically, this snippet:

```
    if run_cmd "$CC -v"; then
      cc_name_tmp=$($CC -v 2>&1 | tail -n 1 | cut -d ' ' -f 1)
    elif run_cmd "$CC --version"; then
      cc_name_tmp=$($CC --version 2>&1 | head -n 1 | cut -d ' ' -f 1)
    fi
    if test -n "${cc_name_tmp}"; then
      if echo "$cc_name_tmp" | grep -q "gcc"; then
        ...
      fi
      if echo "$cc_name_tmp" | grep -q "clang"; then
        start_check "$CC version"
        cc_vendor=clang
        cc_version=$($CC -dumpversion 2>&1)
        finish_check "clang $cc_version"
        break
      fi
```

For your reference, the output of `clang -v`/`clang --version` in macOS:
```
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin20.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

And for `gcc -v`/`gcc --version` in macOS (which is actually just `clang`):
```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin20.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by rhash** [comment]
Thanks for the bug report.

Detection of clang was broken by PR #149. Now it'is fixed by 4dc506066cf1727b021e6352535a8bb315c3f8dc.

**Comment by alebcay** [comment]
Thank you! We'll pick this commit as we await the next version release.

**Comment by justinkb** [comment]
this isn't macos specific, I hit this with clang 12.0.0 on linux too

**Comment by rhash** [comment]
Fix included into [RHash v1.4.3](https://github.com/rhash/RHash/releases/tag/v1.4.3).
